### ISSUE TITLE
Cannot figure out where Portmaster is blocking my iphone USB connection.

### ISSUE THREAD

**Issue by griesi007** [safing/portmaster#1708]
Hi,

currently I have to shut down Portmaster completely to be able to backup my phone using iTunes or iMazing. In both cases the USB connection cannot be established when Portmaster is running. I already alowed connections for Bonjour Service and Windows SSDSRP. But still cannot connect. I do not see anymore related connections which are blocked. Any ideas what to look for?

**Comment by Raphty** [comment]
We are "ignoring" some connection types atm, Like #166

I am not sure how iTunes or iMazing connect to the iPhone, but I can imagine that this runs into similar issues like VMs do.

We are considering tacklinge the whole VM thing after the next bigger releases.

**Comment by griesi007** [comment]
@Raphty Thanks for taking care. I am not sure if I understood you correctly. But isn´t it that way, that if a connection is blocked by Portmaster ist should be listed in the blocked connections and otherwise be untouched and thus work? Or is it possible that Portmaster blocks the connection and there is no trace of it in Portmaster?

**Comment by Raphty** [comment]
You are correct, for "normal" connections Portmaster shows what it does in the network log.

But I assume that the way these apps try to establish a connection is similar to a VM from a network perspective.
At the moment it depends on the implementation of this network connections if Portmaster just displays them as one app (I guess that will be the desired behavior for VM support in the future, VirtualBox and Gnome Boxes work like that) and in other cases Portmaster does not show you the connections (again because they are "ignored" and considered out of scope atm) and either blocks or allows all of them. This is depending on how they are implemented.

**Comment by griesi007** [comment]
@Raphty Thanks for clarification. So there might be no way for me to identify and allow these blocked connections in Portmaster and the only solution is to shutdown Portmaster when doing iphone backups locally? (which is bad). Is there anything I can do to help you analyzing the root cause so this could be adressed in future releases?

**Comment by Raphty** [comment]
thank you for offering to help, we are open for contributions.
If you have experience in networking please reach out and lets figure out the specification.

We on our own will not start working on this in the next months, but we can help you or another FOSS contributer with the specification and will do the code review afterwards.

**Comment by griesi007** [comment]
@Raphty I do not have the knowledge to directly contribute. I was thinking about helping to analyze what causes those connections to be blocked. As I am a developer I have technical knowledge but not in depth regarding networking and digging into network related issues. So I could provide some analysis when being instructed accordingly.

**Comment by Raphty** [comment]
thank you, for offering. I might come back to you when we decide to start tackling this.
